### PR TITLE
Adding Support for NSProgressIndicator

### PR DESCRIPTION
**Pull request by phaus**
For a small application, I needed a progress indicator.
So after some hours, I was able to add this to the code.
I also updated the helloworld example.

### CODE REVIEW

**Comment by mojbro** [comment]
Amazing! I was not aware anyone was using this project. I'll have a closer look at the PR and merge it soon. Hope it works well for your project. This makes me inspired to get back to this project and develop it further. @phaus

**Comment by phaus** [comment]
I am currently refining the UI a little bit more. 
So please wait with this merge - if you plan to do merge :-). 
This project was a great start for my purpose :-).

**Comment by phaus** [comment]
Okay. Done with my updates. The behavior is now like expected :-)

File: button.m [hunk 1]
```diff
@@ -10,8 +10,9 @@ -(void) buttonClicked:(id) sender
 
 ButtonPtr Button_New(int goButtonID, int x, int y, int w, int h) {
     id nsButton = [[[NSButton alloc] initWithFrame:NSMakeRect(x, y, w, h)] autorelease];
-    ButtonHandler* handler = [[ButtonHandler alloc] init]; // TODO: need to be released somewhere
+    ButtonHandler* handler = [[ButtonHandler alloc] init];
     [handler setGoButtonID:goButtonID];
+    [handler autorelease];
```
Comment (mojbro) [inline button.m:15]: Awesome.

File: examples/helloworld/helloworld.go [hunk 2]
```diff
@@ -2,19 +2,26 @@ package main
 
 import (
 	"fmt"
+	"time"
 
-	"github.com/mojbro/gocoa"
+	"github.com/phaus/gocoa"
```
Comment (mojbro) [inline examples/helloworld/helloworld.go]: Could you replace this back to `github.com/mojbro/gocoa` just for this original fork of this project?

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module github.com/phaus/gocoa
```
Comment (mojbro) [inline go.mod]: Could you change this to the `github.com/mojbro/gocoa` url too?

File: examples/helloworld/go.mod [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module helloworld
+
+go 1.12
+
+replace github.com/phaus/gocoa => ../../
```
Comment (mojbro) [inline examples/helloworld/go.mod]: Maybe `github.com/mojbro/gocoa` here

File: examples/helloworld/go.mod [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+require github.com/phaus/gocoa v0.0.0-00010101000000-000000000000 // indirect
```
Comment (mojbro) [inline examples/helloworld/go.mod]: And `github.com/mojbro/gocoa`

File: progressIndicator_test.go [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	indicator.IncrementBy(1.00)
+	if indicator.GetValue() != 5.00 {
+		t.Fatalf("indicator did not increment the value but %f!", indicator.GetValue())
+	}
+
+	indicator.IncrementBy(-1.00)
+	if indicator.GetValue() != 4.00 {
+		t.Fatalf("indicator did not decrement the value but %f!", indicator.GetValue())
+	}
+}
```
Comment (mojbro) [inline progressIndicator_test.go:59]: Wohoo, tests! Great!

**Review by mojbro — CHANGES_REQUESTED** [review]
Hey, this is really awesome. I don't have a Go development environment set up on this computer, so can't test it right now, but looks great to me. The only thing I'd like to change is to change the import paths back to `github.com/mojbro/gocoa` for this repository. Could you help me with that? 

I've been away from Go dev for a while, so I haven't dug into modules yet. Happy to see that coming in here.

Also, I'd like to credit as a contributor you in the README file. I'll add that after merging. I guess you've added about as much to this project as I have.

**Comment by phaus** [comment]
I guess it will be difficult to have correct go dependencies within forks :-).
I set them to this repository ;-).

**Comment by mojbro** [comment]
Looovely!

**Comment by mojbro** [comment]
@phaus I added you as a collaborator if you want to push directly to this repo without having to go through the fork/pull request process. (Although I like the pull request concept for the review stage)

**Comment by phaus** [comment]
I will go with feature branches an pull requests anyway. But doing it within the same repo makes it easier. Thx for the invite :-).

**Pull request merged**